src/main/java/Parser.java

```diff
@@ -0,0 +1,73 @@
+import java.time.LocalDate;
+import java.time.format.DateTimeParseException;
+
+/**
+ * Parses user inputs.
+ */
+public class Parser {
+
+    /**
+     * Throws an error if the given array only has 1 element.
+     * @param task Type of task.
+     * @param hasDescription Whether the task has a description.
+     * @param elements Array to check.
+     * @throws DukeException If given array has only 1 element.
+     */
+    private static void checkArr(String task, boolean hasDescription, String[] elements) throws DukeException {
+        if (elements.length == 1) {
+            throw new MissingInfoException(task, hasDescription);
+        }
+    }
+
+    /**
+     * Parses the given user input and maps it to a command.
+     * @param input User input.
+     * @return Command.
+     * @throws DukeException If any error occurs during parsing of the user input.
+     */
+    public static Command parse(String input) throws DukeException {
+        Command c;
+        String[] inputElements = input.split(" ", 2);
+        String command = inputElements[0];
+        try {
+            if (command.equals("bye")) {

```

Comment:
If an instruction is "bye but not bye" or "list abcd". It is supposed to be interpreted as invalid input. However in this case it is interpreted as valid. A small bug to fix. ;-)